commit 16996a9156c9ff2924bdb19ff43d40617a41c912
Author: Vineetha Kamath <vineetha.hari.pai@canonical.com>
Date:   Tue Jul 23 15:32:32 2019 -0400

From: Vineetha Kamath<vineetha.hari.pai@canonical.com>
Decription: Disable libgcrypt reading /proc/sys/crypto/fips_enabled
file and going into FIPS mode. libnss is not a FIPS
certified library.
Bug-Ubuntu: http://bugs.launchpad.net/bugs/1837734
Forwarded: not-needed
Updated: 2024-03-21

--- a/nss/lib/sysinit/nsssysinit.c
+++ b/nss/lib/sysinit/nsssysinit.c
@@ -171,11 +171,13 @@ static PRBool
 getFIPSMode(void)
 {
 #ifndef NSS_FIPS_DISABLED
-    FILE *f;
+    FILE *f = NULL;
     char d;
     size_t size;
 
+#if 0
     f = fopen("/proc/sys/crypto/fips_enabled", "r");
+#endif
     if (!f) {
         /* if we don't have a proc flag, fall back to the
      * environment variable */
--- a/nss/lib/freebl/stubs.c
+++ b/nss/lib/freebl/stubs.c
@@ -737,10 +737,12 @@ NSS_GetSystemFIPSEnabled_stub(void)
  * as they become available/known */
 #ifdef LINUX
     {
-        FILE *f;
+        FILE *f = NULL;
         char d;
         size_t size;
+#if 0
         f = fopen("/proc/sys/crypto/fips_enabled", "r");
+#endif
         if (!f)
             return PR_FALSE;
 
--- a/nss/lib/util/secport.c
+++ b/nss/lib/util/secport.c
@@ -905,10 +905,12 @@ NSS_GetSystemFIPSEnabled(void)
  * as they become available/known */
 #ifdef LINUX
     {
-        FILE *f;
+        FILE *f = NULL;
         char d;
         size_t size;
+#if 0
         f = fopen("/proc/sys/crypto/fips_enabled", "r");
+#endif
         if (!f)
             return PR_FALSE;
 
